File: GXpert/GXpert.Web/Modules/ServerTypes/Attendance/RaiseHandRecordedSessionColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { RaiseHandRecordedSessionRow } from "./RaiseHandRecordedSessionRow";

export interface RaiseHandRecordedSessionColumns {
    Id: Column<RaiseHandRecordedSessionRow>;
    ContentTitle: Column<RaiseHandRecordedSessionRow>;
    StudentPrn: Column<RaiseHandRecordedSessionRow>;
    PlayListTitle: Column<RaiseHandRecordedSessionRow>;
    PlayListContentId: Column<RaiseHandRecordedSessionRow>;
    ActivationDeviceId: Column<RaiseHandRecordedSessionRow>;
}

export class RaiseHandRecordedSessionColumns extends ColumnsBase<RaiseHandRecordedSessionRow> {
    static readonly columnsKey = 'Attendance.RaiseHandRecordedSession';
    static readonly Fields = fieldsProxy<RaiseHandRecordedSessionColumns>();
}

File: GXpert/GXpert.Web/Modules/Attendance/RaiseHandRecordedSession/RaiseHandRecordedSessionColumns.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace GXpert.Attendance.Columns;

[ColumnsScript("Attendance.RaiseHandRecordedSession")]
[BasedOnRow(typeof(RaiseHandRecordedSessionRow), CheckNames = true)]
public class RaiseHandRecordedSessionColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    public string ContentTitle { get; set; }
    public string StudentPrn { get; set; }
    public string PlayListTitle { get; set; }
    public int PlayListContentId { get; set; }
    public string ActivationDeviceId { get; set; }
}